providers/aws/resources/aws_ec2.go

```diff
@@ -1787,10 +1787,12 @@ func shouldExcludeInstance(instance ec2types.Instance, filters connection.Ec2Dis
 		}
 	}
 	for k, v := range filters.ExcludeTags {
-		for _, iTag := range instance.Tags {
-			if iTag.Key != nil && *iTag.Key == k &&
-				iTag.Value != nil && *iTag.Value == v {
-				return true
+		for _, tagValue := range strings.Split(v, ",") {
+			for _, iTag := range instance.Tags {
+				if iTag.Key != nil && *iTag.Key == k &&
+					iTag.Value != nil && *iTag.Value == tagValue {
+					return true
+				}
```

Comment:
nit: doesn't this for block essentially check whether `filters.ExcludeTags` and `instance.Tags` intersect? I'm not go expert, but there must be a built-in function for that ❓

Reply:
I feel like in Go the idiomatic way to do most things with maps/slices is just for loops. But even if there was such a function, we are expanding the values of `filters.ExcludeTags` which can be comma-separated so it wouldn't apply here.